discord_bots/utils.py

```diff
@@ -528,54 +528,42 @@ async def print_leaderboard(channel=None):
                 player: Player = (
                     session.query(Player).filter(Player.id == pct.player_id).first()
                 )
-                output += f"\n{i}. {round(pct.rank, 1)} - {player.name} _(mu: {round(pct.mu, 1)}, sigma: {round(pct.sigma, 1)})_"
-            output += "\n"
+                output += f"\n{i}. {round(pct.rank, 1)} - <@{player.id}> _(mu: {round(pct.mu, 1)}, sigma: {round(pct.sigma, 1)})_"
         pass
-    else:
-        output = "**Leaderboard**"
-        top_40_players: list[Player] = (
+    
+    if config.ECONOMY_ENABLED:
+        output += f"\n\n**{config.CURRENCY_NAME}**"
+        top_10_player_currency: list[Player] = (
             session.query(Player)
-            .filter(Player.rated_trueskill_sigma != 5.0)  # Season reset
-            .filter(Player.rated_trueskill_sigma != 12.0)  # New player
-            .filter(Player.leaderboard_enabled == True)
-            # .order_by(Player.leaderboard_trueskill.desc())
-            # .limit(20)
+            .order_by(Player.currency.desc())
+            .limit(10)
         )
-        players_adjusted = sorted(
-            [
-                (player.rated_trueskill_mu - 3 * player.rated_trueskill_sigma, player)
-                for player in top_40_players
-            ],
-            reverse=True,
-        )[0:50]
-        for i, (_, player) in enumerate(players_adjusted, 1):
-            output += f"\n{i}. {round(player.leaderboard_trueskill, 1)} - {player.name} _(mu: {round(player.rated_trueskill_mu, 1)}, sigma: {round(player.rated_trueskill_sigma, 1)})_"
-    session.close()
+        for i, player_currency in enumerate(top_10_player_currency, 1):
+            output += f"\n{i}. {player_currency.currency} - <@{player_currency.id}>"
+    
+    output += "\n"
     output += "\n(Ranks calculated using the formula: _mu - 3*sigma_)"
     output += "\n(Leaderboard updates periodically)"
     output += "\n(!disableleaderboard to hide yourself from the leaderboard)"
+    
+    session.close()
 
     if config.LEADERBOARD_CHANNEL:
-        leaderboard_channel = bot.get_channel(config.LEADERBOARD_CHANNEL)
+        leaderboard_channel: TextChannel = bot.get_channel(config.LEADERBOARD_CHANNEL)
```

Comment:
pylance will complain with this type hint, since you're assuming that a `TextChannel` will be returned (which it will be 99% of the time), to get around this I just do:
```python
leaderboard_channel = bot.get_channel = bot.get_channel(config.LEADERBOARD_CHANNEL)
if isinstance(leaderboard_channel, discord.TextChannel):
    # note: isinstance handles None for us
    ...
```
This isn't a big deal, but I'd prefer if we didn't add anything that made pylance complain

Reply:
Fair point, changed things around a bit to clear up the typing errors
[64f989c](https://github.com/tribesonecommunity/discord-bots/pull/65/commits/64f989cfdc0157357485970dab481e0b31f5b0bb)